vexl/Repositories/OfferRepository.swift

```diff
@@ -203,6 +204,11 @@ class OfferRepository: OfferRepositoryType {
         persistence.load(type: ManagedOffer.self, context: persistence.viewContext, predicate: NSPredicate(format: "offerID != nil"))
     }
 
+    func getKnownOffersThatAreNotMine() -> AnyPublisher<[ManagedOffer], Error> {
+        persistence.load(type: ManagedOffer.self,
+                         context: persistence.viewContext, predicate: NSPredicate(format: "offerID != nil AND adminID == nil"))
```

Comment:
While this condition gives you correct result too, I generally managed this kind of source of truth using a relationship with other entities. In this case you should be asking which offers are in relationship with the ManagedUser  
```suggestion
                         context: persistence.viewContext, predicate: NSPredicate(format: "user == nil"))
```